card/mod/core/set/all/fetch_helper.rb

```diff
@@ -72,13 +72,16 @@ def retrievable_mark_type_and_value mark
     end
   end
 
-  def retrieve_from_db? card, opts
-    card.nil? || (opts[:look_in_trash] && card.new_card? && !card.trash)
+  def retrieve_from_db? card, mark_type, mark_key, look_in_trash
+    return false if card&.real?                                      # real card found in cache
+    return false if mark_type === :key && mark_key.to_name.relative? # no relative names in db
+    return true if card.nil?                                         # card not found in cache
```

Comment:
a minor style issue: I think `card.nil?` is the main case that this method  covers but with the return statement it became more like a exceptional  state.